src/v/cluster/data_migration_table.cc

```diff
@@ -32,7 +32,7 @@ migrations_table::migrations_table(
 bool migrations_table::is_valid_state_transition(state current, state target) {
     switch (current) {
     case state::planned:
-        return target == state::preparing;
+        return target == state::preparing || target == state::canceling;
```

Comment:
isn't that enough to delete the planned migrartion without first cancelling it ? If it is not can you update the state tranistion diagram ?

Reply:
The state transition diagram in `src/v/cluster/data_migration_types.h` allows planned->canceling transition. I agree it's better to delete though, so I'll remove this change and change the diagram instead.